rust/cardano-chain-follower/src/turbo_downloader/mod.rs

```diff
@@ -298,21 +307,21 @@ impl ParallelDownloadProcessorInner {
 
     /// Get start offset of a chunk.
     fn chunk_start(&self, chunk: usize) -> usize {
-        self.cfg.chunk_size * chunk
+        self.cfg.chunk_size.saturating_mul(chunk)
     }
 
     /// Get inclusive end offset of a chunk.
     fn chunk_end(&self, chunk: usize) -> usize {
         let start = self.chunk_start(chunk);
-        if start + self.cfg.chunk_size >= self.file_size {
-            self.file_size - 1
+        if start.saturating_add(self.cfg.chunk_size) >= self.file_size {
+            self.file_size.saturating_sub(1)
         } else {
-            start + self.cfg.chunk_size - 1
+            start.saturating_add(self.cfg.chunk_size).saturating_sub(1)
         }
     }
 
     /// Sends a GET request to download a chunk of the file at the specified range
-    fn get_range(&self, agent: &ureq::Agent, chunk: usize) -> anyhow::Result<Arc<Vec<u8>>> {
+    fn get_range(&self, agent: &ureq::Agent, chunk: usize) -> Result<Arc<Vec<u8>>> {
```

Comment:
To avoid confusion that this could be a standard `Result` type, I would leave a an explicit namespace of this `anyhow::Result`

Reply:
Discussed: `anyhow::Result` is already [imported](https://github.com/input-output-hk/catalyst-libs/blob/main/rust/cardano-chain-follower/src/turbo_downloader/mod.rs#L21) and [used without namespace](https://github.com/input-output-hk/catalyst-libs/blob/main/rust/cardano-chain-follower/src/turbo_downloader/mod.rs#L48) in this file, so we decided to keep things consistent and remove the namespace.